src/behavior.ts

```diff
@@ -74,17 +77,20 @@ export class Behavior {
         });
     }
 
-    private select(dataPoints: SelectableDataPoint | SelectableDataPoint[], multiSelect: boolean): void {
+    private get hasSelection(): boolean {
+        const selectionIds = this.selectionManager.getSelectionIds();
+        return selectionIds.length > 0;
+    }
+
+    private select(dataPoints: SelectableDataPoint, multiSelect: boolean): void {
         if (!dataPoints) {
             return;
         }
 
-        if (!Array.isArray(dataPoints)) {
-            dataPoints = [dataPoints];
-        }
+        const arrayDataPoints = [dataPoints];
 
         const selectionIdsToSelect: ISelectionId[] = [];
-        for (const dataPoint of dataPoints) {
+        for (const dataPoint of arrayDataPoints) {
```

Comment:
you don't need a loop if you always have only 1 datapoint

Reply:
refactored